pkg/flare/archive.go

```diff
@@ -393,30 +393,14 @@ func zipConfigFiles(tempDir, hostname string, confSearchPaths SearchPaths, perms
 	if config.Datadog.ConfigFileUsed() != "" {
 		// zip up the config file that was actually used, if one exists
 		filePath := config.Datadog.ConfigFileUsed()
-
-		// Check if the file exists
-		_, err := os.Stat(filePath)
-		if err == nil {
-			f = filepath.Join(tempDir, hostname, "etc", "datadog.yaml")
-			err := ensureParentDirsExist(f)
-			if err != nil {
-				return err
-			}
-
-			w, err := newRedactingWriter(f, os.ModePerm, true)
-			if err != nil {
-				return err
-			}
-			defer w.Close()
-
-			_, err = w.WriteFromFile(filePath)
-			if err != nil {
-				return err
-			}
-
-			if permsInfos != nil {
-				permsInfos.add(filePath)
-			}
+		if err = createConfigFiles(filePath, tempDir, hostname, permsInfos); err != nil {
+			return err
+		}
+		// figure out system-probe file path based on main config path,
+		// and use best effort to include system-probe.yaml to the flare
+		systemProbePath := getSystemProbePath(filePath)
+		if systemErr := createConfigFiles(systemProbePath, tempDir, hostname, permsInfos); systemErr != nil {
+			log.Warnf("could not zip system-probe.yaml: %s", systemErr)
```

Comment:
I'm fine with this log level, but let's make it clear that the user might not have configured the `system-probe`.